streams/src/main/java/org/apache/kafka/streams/kstream/internals/StatefulProcessorNode.java

```diff
@@ -0,0 +1,133 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one or more
+ * contributor license agreements. See the NOTICE file distributed with
+ * this work for additional information regarding copyright ownership.
+ * The ASF licenses this file to You under the Apache License, Version 2.0
+ * (the "License"); you may not use this file except in compliance with
+ * the License. You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.apache.kafka.streams.kstream.internals;
+
+import org.apache.kafka.streams.processor.ProcessorSupplier;
+import org.apache.kafka.streams.processor.StateStoreSupplier;
+import org.apache.kafka.streams.processor.internals.InternalTopologyBuilder;
+import org.apache.kafka.streams.state.KeyValueStore;
+import org.apache.kafka.streams.state.StoreBuilder;
+
+import java.util.Arrays;
+
+class StatefulProcessorNode<K, V> extends StatelessProcessorNode<K, V> {
+
+    private final String[] storeNames;
+    private final org.apache.kafka.streams.processor.StateStoreSupplier<KeyValueStore> storeSupplier;
+    private final StoreBuilder<KeyValueStore<K, V>> storeBuilder;
+
+
+    StatefulProcessorNode(final String predecessorNodeName,
+                          final String name,
+                          final ProcessorSupplier processorSupplier,
+                          final String[] storeNames,
```

Comment:
What if we do have multiple storeNames? Since we only have one `storeBuilder`, that seems not working if we assume one builder per store name?

Reply:
The multiple store names come from the `process` and `transformX` methods where users can supply N number of state store names previously created with `builder.addStateStore` 

Do we want to consider another type of stateful node? WDYT?

Reply:
ack